### DISCUSSION TITLE
DnD 5e Character Sheet

### DISCUSSION THREAD

**Discussion by Audio951** [mProjectsCode/obsidian-meta-bind-plugin#32] [Show and tell]
Hello! I'm currently working on a character sheet setup for my small group to use and figured others might like to see and use it once it's complete. I'll post progress updates here for now as I work on it and am open to feedback!

It essentially is a small collection of community plugins, a ton of css, and a little javascript. In order to not overthink the layout too much, I followed the basic layout of a normal DnD 5e character sheet and reorganized it to be more suitable for landscape viewing. I think it's a little busy and hard to parse at a glance right now, but it's coming along.

Here's the first section:
![image](https://user-images.githubusercontent.com/9119761/204963755-ecf3f9a7-9785-48c3-a16a-4dc0b84074de.png)

Here's the second section:
![image](https://user-images.githubusercontent.com/9119761/204963803-56f90ab0-582f-43f1-a624-a155650f8415.png)

And here's where the third section would be - IF I HAD ONE:
(coming soon)

The primary motivation was being able to hover over linked notes and see a preview of them - it makes combat so much more smooth having quick reference to skills and abilities:
![example1](https://user-images.githubusercontent.com/9119761/204967441-4cd2fa9b-6981-443c-aa2f-79a74430fd3d.gif)

Additionally, we can use Dataview inline javascript to calculate stuff:
![example2](https://user-images.githubusercontent.com/9119761/204967447-b721cf35-11ec-4722-8017-a1970eab33df.gif)

One downside is there's currently a delay to the updating that can rollback changes made using the interactive elements.
I'll take a stab at reducing the impact of that once I have it put together and looking decent.
There's also some weirdness with getting the javascript to run since obsidian seems to only process what is currently being rendered in the view, so inline javascript via Dataview is only consistent if I put it at the bottom and scroll down in edit mode so it has a chance to run.

This is my first real delve into css/html so it's been a ton of learning and rewriting everything multiple times, but I'm fairly happy with where it's going so far.

**Comment by mayurankv** [comment]
Are you likely to post a template at any point? What plugins are you using?

**Reply by Audio951** [reply]
Yeah, I'll post it somewhere once I think it's feature-complete. 

The plugins are CustomJS, Dataview, Dice Roller (maybe), and of course Meta Bind.

**Reply by sundeer** [reply]
It's been a while but have you shared this anywhere? I'd like to see it.

**Reply by Devilofether** [reply]
How did you avoid using JS engine?, I want to know because it won't install properly for me; and I prefer the simplicity of CustomJS

**Comment by Audio951** [comment]
Third page complete:
![image](https://user-images.githubusercontent.com/9119761/205422041-b377c681-c074-4444-8476-3644a258cf14.png)

I've updated the other two pages as well to make them a tad easier on the eyes:
![image](https://user-images.githubusercontent.com/9119761/205422152-2def8396-6635-4793-a935-1bf84f109f92.png)

![image](https://user-images.githubusercontent.com/9119761/205422217-156e6c6e-7209-4baa-b4ea-f17365b1c74d.png)

Also, I'm still experimenting with how to jank the code into place that modifies stuff so it all works. My current implementation simply has a shiny button at the top which you click once to apply the modifications since I can't get it to work consistently any other way. There are still some issues with editing and being kicked out ect, but once I get that nailed down a bit better it'll probably be ready for a first release.

**Reply by mayurankv** [reply]
There is a dataview command to forces refresh of all blocks!

**Comment by beesinblenders** [comment]
Whoa! I'm in the process of building a character sheet using Meta-Bind and I feel discouraged to see how far this is going. Can you share more detail? 

I can see that inventory items seem to link to specific notes (like 1 object = 1 note). I was going for a more contained strategy using nested YAML fields, like so:
- inventory:
  - weapons:
    - rapier
  - riches:
    - GP: 120
    - stones:
      - emerald

But I can't get my meta-bind queries to target these nested fields efficiently... Do you have any advice? Should I switch to a note-based inventory at once?

Anyway congratulations on this job.

**Comment by Tejeev** [comment]
@Audio951 did you ever publish this template? Even if it's not 100% I would find it very helpful embarking on my own char sheet creation journey.
Thank you for the teaser all those years ago! It's inspiring!

**Comment by VikingInOrbit** [comment]
Hi, are you still working on this? Or is it published somewhere?

**Reply by MrYod** [reply]
I'm subscribed to this thread for info as well.

**Comment by skeetcha** [comment]
Do you have the source for this? Would love to use this (and also develop a PF2e version of it)?

**Comment by pguilimaf** [comment]
Woah, I'd love to have a sheet like that. So organized!

**Comment by Cafrienzy** [comment]
@Audio951 Awesome work!! Wow! Would love the source on this. Even if it’s not finished for you, looks like its exactly what I would need =D
